### PR TITLE
pyusb backend: add exception for Hexiwear

### PR DESCRIPTION
**Pull request by aurabindo**
Enable pyOCD to run as normal user, a hack for Hexiwear.
Without this hack, only root user can debug/flash.
Kinetis Design Studio launches pyOCD as normal user.

Signed-off-by: Aurabindo J <mail@aurabindo.in>

### CODE REVIEW

**Comment by aurabindo** [comment]
This fixes #259

File: pyOCD/pyDAPAccess/interface/pyusb_backend.py [hunk 1]
```diff
@@ -86,6 +86,9 @@ def getAllConnectedInterface():
                 # This can cause an exception to be thrown if the device
                 # is malfunctioning.
                 product = board.product
+            except ValueError:
+                if board.idVendor is 0xd28 and board.idProduct is 0x204:
```
Comment (c1728p9) [inline pyOCD/pyDAPAccess/interface/pyusb_backend.py]: So this code is ignoring exceptions that occur on boards with a vid,pid other than 0xd28, 0x204? Can you provide any more details on this? Which VID/PID does Hexiwear have?

**Comment by aurabindo** [comment]
The intention is to ignore the exception for the specified VID/PID, which belongs to the baseboard, which is part of Hexiwear IoT Kit. This board provides extension ports and most importantly an OpenSDA debug adapter. I've been discussing with pyusb devs regarding [this](this), since its  not easy for me to make eclipse (Kinetis Design Studio) run pyOCD as root, and they dont seem to be interested in [fixing the issue ](nxp)

[issue]:https://sourceforge.net/p/pyusb/mailman/pyusb-users/thread/GwV3hDpxIGyh5ucEH76ywnJIc84EHGdM2HHykpY42146%40mailpile/#msg35297171
[nxp]:https://community.nxp.com/thread/432509

**Comment by c1728p9** [comment]
@aurabindo, I can't get the links in your last post to work. Could you provide updated links?

**Comment by aurabindo** [comment]
@c1728p9 github seems to be doing something nasty. Try these:

1: https://sourceforge.net/p/pyusb/mailman/pyusb-users/thread/CAOsEZohWUs%3D6i7%2BQEz36pFvX3JYiDiu6s_3AAv5qTPzsY2zkWg%40mail.gmail.com/#msg35291047
2: https://community.nxp.com/thread/432509

**Comment by rsalveti** [comment]
Just proposed #262 and saw this pull request. With #262 I'm generically catching valueerror, which can happen with any device (since it probes all the devices you have in your system).

That + a udev file for your own device should be enough.

Happy to close that and update yours with a more generic solution.

**Comment by aurabindo** [comment]
I was not sure if this is specific to hexiwear, which is why I didn't go for a generic solution as upstream might not like it. I am still expecting a comment from @c1728p9

Feel free to update mine if you feel that is appropriate. But you have merge rights, and you think a generic solution makes more sense, please go ahead and merge yours.

**Comment by c1728p9** [comment]
Hi @aurabindo, sorry for not responding sooner. It looks like this is problem for boards in general and not specific to the Hexiwear, so the generic solution is probably more appropriate.

**Comment by aurabindo** [comment]
Thanks for the update @c1728p9. #262 will be more appropriate then

**Pull request closed without merging**